### PR TITLE
[DCOS-39150] cluster wide unique ids for spark executors

### PR DESCRIPTION
**Pull request by akirillov**
This change adds test `test_unique_task_ids` which submits two spark jobs and verifies that Mesos task ids (which are also Spark Executor ids) are different for each driver. Previously, each driver assigned integer ids starting from `0` which led to similar task ids for executors belonging to different drivers. `test_task_not_lost` test is slightly modified to address the switch from numeric ids to `UUID` with executor index suffix.

### CODE REVIEW

File: tests/test_spark.py [hunk 1]
```diff
@@ -353,6 +354,47 @@ def test_unique_vips():
...
+                                        args=["--class org.apache.spark.examples.SparkPi"])
+    driver_id_2 = utils.submit_job(app_url=utils.SPARK_EXAMPLES,
+                                        app_args="100",
+                                        args=["--class org.apache.spark.examples.SparkPi"])
+
+    LOGGER.info('Two Spark Applications submitted. Driver 1 ID: %s, Diver 2 ID: %s'%(driver_id_1,driver_id_2))
+    LOGGER.info('Waiting for completion. Polling state')
+    completed = False
+    timeout_sec = 300
+    while timeout_sec > 0:
```
[the remark this answers is not in the document]
Comment (akirillov) [reply tests/test_spark.py]: done

File: tests/test_spark.py [hunk 2]
```diff
@@ -353,6 +354,47 @@ def test_unique_vips():
...
+def test_unique_task_ids():
+    LOGGER.info('Submitting two sample Spark Applications')
+    driver_id_1 = utils.submit_job(app_url=utils.SPARK_EXAMPLES,
+                                        app_args="100",
+                                        args=["--class org.apache.spark.examples.SparkPi"])
+    driver_id_2 = utils.submit_job(app_url=utils.SPARK_EXAMPLES,
+                                        app_args="100",
+                                        args=["--class org.apache.spark.examples.SparkPi"])
+
+    LOGGER.info('Two Spark Applications submitted. Driver 1 ID: %s, Diver 2 ID: %s'%(driver_id_1,driver_id_2))
```
[the remark this answers is not in the document]
Comment (akirillov) [reply tests/test_spark.py]: fixed

File: manifest.json [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {
     "spark_version": "2.2.1",
     "default_spark_dist": {
-        "hadoop_version": "2.6",
-        "uri": "https://downloads.mesosphere.com/spark/assets/spark-2.2.1-3-SNAPSHOT-bin-2.6.tgz"
+        "hadoop_version": "2.7",
+        "uri": "https://s3.us-east-2.amazonaws.com/akirilloiv-dev/spark-2.2.1-bin-2.7.3.tgz"
```
Comment (samvantran) [inline manifest.json]: Before you merge, pls change this url to a `downloads.mesosphere` spark one. You can append `-SNAPSHOT` to denote that it is a dev build and subject to change

Also, is this spark 2.2.1? Wasn't it built from Spark-2.3?

Comment (akirillov) [reply manifest.json]: The change started from 2.2.1 branch and already merged there, there's a separate PR for Spark 2.3 branch.

Updated versions of Spark distributions are uploaded to `downloads.mesosphere` for both Hadoop 2.6 and 2.7, will wait for the build to succeed and merge the changes after that if it's ok.

**Pull request merged**